tests/inverted-index-test.js

```diff
@@ -1,49 +1,217 @@
+        }
+      ];
+
+      const newTextTitle = 'alice in the wonder land i am not as lazy as alice lady the lady is 50 years old. she has a child';
+      newBook = InvertedIndex.joinTextTitle(newBook);
+      expect(newTextTitle).toEqual(newBook);
+    });
+  });
+
+  describe('Remove duplicate words function', () => {
+    it('should return only unique words in an array', () => {
+      let getWords =
+        ['broken',
+          'colour',
+          'crayons',
+          'go',
+          'go',
+          'he',
+          'he',
+          'is',
+          'is',
+          'poem',
+          'premier',
+          'queen',
+          'still',
+          'to',
+          'to',
+          'up',
+          'up',
+          'yes'];
+
+      const uniqueWords =
+        ['broken',
+          'colour',
+          'crayons',
+          'go',
+          'he',
+          'is',
+          'poem',
+          'premier',
+          'queen',
+          'still',
+          'to',
+          'up',
+          'yes'];
+
+      getWords = InvertedIndex.removeDuplicates(getWords);
+      expect(uniqueWords).toEqual(getWords);
+    });
+  });
+
+  describe('Tokenize function', () => {
+    it('should return in an array tokens for each file', () => {
+      // Parameter to be used to test for correctness of the tokenize method
+      let newBook = [ 
+        {
+          'title': 'Alice in Wonderland',
+          'text': 'I am not as lazy as Alice'
+        },
+        {
+          'title': 'Lady',
+          'text': 'The lady is 50 years old. SHe has a child'
+        }
+      ];
+
+      // Expected result after newBook is passed into tokenize function
+      const result =
+        ['50',
+          'a',
+          'alice',
+          'am',
+          'as',
+          'child',
+          'has',
+          'i',
+          'in',
+          'is',
+          'lady',
+          'lazy',
+          'not',
+          'old',
+          'she',
+          'the',
+          'wonderland',
+          'years'];
+
+      newBook = InvertedIndex.tokenize(newBook);
+      expect(result).toEqual(newBook);
     });
-    it('should return \'Invalid input\' for non-integers', () => {
-      expect('day').toEqual('Invalid input');
+  });
+
+  describe('Create Index function', () => {
+    it('should return error message if JSON file is `Invalid`, `Malformed` or `empty` file.', () => {
+      expect(invObj.createIndex(invalidFile)).toEqual('Error: The file is not a correct JSON file!');
     });
-    it('should return \'Invalid input\' for non-integers', () => {
-      expect('-').toEqual('Invalid input');
+
+    it('should return in an object a file name as key, and accurate index for every word\'s  occurence in a book', () => {
+      const newFile = invObj.createIndex('validFile', validFile);
+      expect(createdIndex).toEqual(newFile);
     });
-    it('should return \'Invalid input\' for non-integers', () => {
-      expect('/').toEqual('Invalid input');
+  });
+
```

Comment:
Can you show me the tests for searchIndex that ensure passed in index is in the valid format, searchIndex can handle an array of search terms, searchIndex can handle a varied number of search terms as arguments, searchIndex goes through all indexed files if a filename/key is not passed?